test/Microsoft.Azure.WebJobs.Host.EndToEndTests/Scale/ScaleHostEndToEndTests.cs

```diff
@@ -233,27 +235,38 @@ public ITargetScaler GetTargetScaler()
             }
         }
     }
-    public static class TestExtensionAHostBuilderExtensions
+
+    public class TestExtensionA : IWebJobsStartup2
     {
-        public static IHostBuilder ConfigureTestExtensionAScale(this IHostBuilder builder, TriggerMetadata triggerMetadata)
+        public void Configure(IWebJobsBuilder builder)
+        {
+        }
+
+        public void Configure(WebJobsBuilderContext context, IWebJobsBuilder builder)
         {
-            builder.ConfigureServices((context, services) =>
+            var scaleHostBuilderContext = context as ScaleHostBuilderContext;
+            if (scaleHostBuilderContext != null)
             {
-                services.AddSingleton<IScaleMonitorProvider>(serviceProvider =>
+                foreach (var triggerMetadata in scaleHostBuilderContext.GetTriggersMetadata(ATriggerType))
                 {
-                    IConfiguration config = serviceProvider.GetService<IConfiguration>();
-                    IOptions<ScaleOptions> scaleOptions = serviceProvider.GetService<IOptions<ScaleOptions>>();
-                    return new TestExtensionAScalerProvider(config, scaleOptions, triggerMetadata);
-                });
-                services.AddSingleton<ITargetScalerProvider>(serviceProvider =>
-                {
-                    IConfiguration config = serviceProvider.GetService<IConfiguration>();
-                    IOptions<ScaleOptions> scaleOptions = serviceProvider.GetService<IOptions<ScaleOptions>>();
-                    return new TestExtensionAScalerProvider(config, scaleOptions, triggerMetadata);
-                });
-            });
-
-            return builder;
+                    builder.Services.AddSingleton<IScaleMonitorProvider>(serviceProvider =>
+                    {
+                        IConfiguration config = serviceProvider.GetService<IConfiguration>();
+                        IOptions<ScaleOptions> scaleOptions = serviceProvider.GetService<IOptions<ScaleOptions>>();
+                        return new TestExtensionAScalerProvider(config, scaleOptions, triggerMetadata);
```

Comment:
I think we're really close to being able to clean all this up so scaler providers can be registered as a Type, rather than having to construct explicitly this way.

If we introduce a new `ITriggerMetadataProvider` interface, the host can register that with DI to provide the metadata, then this extension logic simply becomes the following, allowing DI to be fully used for construction:

```csharp
if (scaleHostBuilderContext != null)
{
    builder.Services.AddSingleton<IScaleMonitorProvider, TestExtensionAScalerProvider>();
    builder.Services.AddSingleton<ITargetScalerProvider, TestExtensionAScalerProvider>();
}
```

Reply:
`IScaleMonitorProvider/ITargetScalerProvider` returns singular `IScaleMonitor `and `ITargetScaler`. It looks like we need a some new interface to return collection of scalers in this case.

Reply:
My latest commit addresses this